content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,169 @@
+---
+kind: tutorial
+languages: [javascript]
+sidebar: "language:javascript"
+prev: /languages/javascript/codewars-test/
+---
+
+# JavaScript: creating and translating a kata
+
+This article is meant as a help for kata authors and translators who would like to create new content in JavaScript. It attempts to explain how to create and organize things in a way conforming to [authoring guidelines](/authoring/guidelines/), shows the most common pitfalls and how to avoid them.
+
+This article is not a standalone tutorial on creating kata or translations. It's meant to be a complementary, JavaScript-specific part of a more general set of HOWTOs and guidelines related to [content authoring](/authoring/). If you are going to create a JavaScript translation, or a new JavaScript kata from scratch, please make yourself familiar with the aforementioned documents related to authoring in general first. 
+
+
+## General info
+
+Any technical information related to the JavaScript setup on Codewars can be found on the [JavaScript reference](/languages/javascript/) page (language versions, available libraries, and setup of the code runner).
+
+## Description
+
+JavaScript code blocks can be inserted with JavaScript-specific part in [sequential code blocks](/references/markdown/extensions/#sequential-code-blocks):
+
+~~~
+```javascript
+
+...your code here...
+
+```
+~~~
+
+JavaScript-specific paragraphs can be inserted with [language conditional rendering](/references/markdown/extensions/#conditional-rendering):
+
+```
+~~~if:javascript
+
+...text visible only for JavaScript description...
+
+~~~
+
+~~~if-not:javascript
+
+...text not visible in JavaScript description...
+
+~~~
+```
+
+## Tasks and Requirements
+
+Some concepts don't always translate well to or from JavaScript. Because of this, some constructs should be avoided and some translations just shouldn't be done.
+- Avoid returning different data types depending on the situation (_"Return the result, or the string 'Error' if no result can be found..."_). JavaScript is dynamically typed, which is not the case for some other languages. Returning `null` might be appropriate in some situations, but throwing an error might be better in others.
+
+_TODO: Finish this_
+
+## Tests
+
+### Testing framework
+
+:::note
+For simplicity, through the rest of this article it's assumed that the `chai` package is imported and `assert` is extracted from it with the following statement:
+
+```javascript
+const chai = require('chai');
+const assert = chai.assert;
+```
+
+This is a convention used in many JavaScript kata, but it's not a requirement, and authors can choose to import the package in any way they find suitable for them. Also, the `should` or `expect` style can also be used from `chai`.
+:::
+
+JavaScript kata should use [Chai](https://www.chaijs.com/) to implement and execute tests. The most commonly used assertions include:
```

Comment:
Chai is an assertion library. The test framework is [Mocha](https://mochajs.org/).